internal/provider/search_attribute_resource.go

```diff
@@ -0,0 +1,292 @@
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Search Attribute Name",
+				Required:            true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "Search Attribute Indexed Value Type, which defines the type of data stored in the attribute",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("Unspecified", "Text", "Keyword", "Int", "Double", "Bool", "Datetime", "KeywordList"), // Ensure only valid types are used
+				},
+			},
+			"namespace": schema.StringAttribute{
+				MarkdownDescription: "Namespace with which the Search Attribute is associated",
+				Optional:            true,
+				Default:             stringdefault.StaticString("default"),
+				Computed:            true,
+			},
+		},
+	}
+}
+
+// Configure sets up the search attribute resource configuration.
+func (r *SearchAttributeResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	tflog.Info(ctx, "Configuring Temporal Search Attribute Resource")
+
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(grpc.ClientConnInterface)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected grpc.ClientConnInterface, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+	tflog.Info(ctx, "Configured Temporal Search Attribute client", map[string]any{"success": true})
+}
+
+// Create is responsible for creating a new search attribute in Temporal.
+func (r *SearchAttributeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data SearchAttributeResourceModel
+
+	client := operatorservice.NewOperatorServiceClient(r.client)
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create attribute
+	indexedValueType, _ := enums.IndexedValueTypeFromString(data.Type.ValueString())
+
+	request := &operatorservice.AddSearchAttributesRequest{
+		Namespace:        data.Namespace.ValueString(),
+		SearchAttributes: map[string]enums.IndexedValueType{data.Name.ValueString(): indexedValueType},
+	}
+	_, err := client.AddSearchAttributes(ctx, request)
+
+	if err != nil {
+		if _, ok := err.(*serviceerror.AlreadyExists); ok {
+			resp.Diagnostics.AddError("Request Error", "Search attribute with that name is already registered: "+err.Error())
+			return
+		}
+		resp.Diagnostics.AddError("Request error", "Search attribute creation failed: "+err.Error())
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("The search attribute: %s of type %s is successfully created", data.Name, data.Type.String()))
+	tflog.Trace(ctx, "Created a resource")
```

Comment:
is this trace log line really needed since we are logging everything in the previous `log.Info` line?